src/main/java/com/ss/training/lms/controller/AdminBranchController.java

```diff
@@ -11,7 +23,101 @@
 @RestController
 public class AdminBranchController {
 
+	/**
+	 * @param branch
+	 * @return
+	 */
+	@PostMapping(path = "/lms/admin/branch")
+	public ResponseEntity<LibraryBranch> createBranch(@RequestBody LibraryBranch branch) {
+		HttpStatus status = HttpStatus.BAD_REQUEST;
+		if (branch == null || (branch.getBranchName() != null && branch.getBranchName().length() > 45)
+				|| (branch.getBranchAddress() != null && branch.getBranchAddress().length() > 45))
+			return new ResponseEntity<LibraryBranch>(branch, status);
+		try {
+			service.createBranch(branch); // this will also set the ID of this branch object if successful
+			status = HttpStatus.CREATED;
+		} catch (ClassNotFoundException | SQLException e) {
+			status = HttpStatus.INTERNAL_SERVER_ERROR;
+		}
+		return new ResponseEntity<LibraryBranch>(branch, status);
+	}
+
 	@Autowired
 	AdminBranchService service;
 
+	/**
+	 * <<<<<<< HEAD
+	 * 
+	 * @return
+	 */
+	@RequestMapping(path = "/lms/admin/branch")
+	public ResponseEntity<List<LibraryBranch>> readBranches() {
+		List<LibraryBranch> branches = null;
+		HttpStatus status = HttpStatus.OK;
+		try {
+			branches = service.readBranches();
+			if (branches == null) // no branches exist in the database
+				status = HttpStatus.NO_CONTENT;
+		} catch (ClassNotFoundException | SQLException e) {
+			status = HttpStatus.INTERNAL_SERVER_ERROR;
+		}
+		return new ResponseEntity<List<LibraryBranch>>(branches, status);
+	}
+
+	/**
+	 * @param id
+	 * @return
+	 */
+	@DeleteMapping(path = "/lms/admin/branches/{id}")

```

Comment:
This should not have the same path as the delete path. Could add delete after the id to distinguish them